runtime.md

```diff
@@ -50,10 +50,14 @@ If a hook returns a non-zero exit code, then an error is logged and the remainin
         "poststop": [
             {
                 "path": "/usr/sbin/cleanup.sh",
-                "args": ["-f"]
+                "args": ["cleanup.sh", "-f"]
             }
         ]
     }
 ```
 
-`path` is required for a hook. `args` and `env` are optional.
+`path` is required for a hook.
+`args` and `env` are optional.
+`args` includes the command as `args[0]`.
+If the `args` field is missing, empty, or `null`, the runtime will use `[path]`.
```

Comment:
On Wed, Aug 26, 2015 at 06:16:33PM -0700, Lai Jiangshan wrote:
> > +If the `args` field is missing, empty, or `null`, the runtime will use `[path]`.
> 
> I think we should use `[]` instead of `null`.  `null` is
> confusing/misleading here, and `"args" : null` is invalided in the
> config.

The docs are about the JSON, where ‘null’ should be a non-confusing
value.  I'd intended “empty” to mean ‘[]’ and “missing” to mean “no
‘args’ key in the object.”.  I think all of those should be mapped to
‘[path]’, although if someone else felt very strongly about it I'd be
ok with ‘[]’ being mapped to ‘[]’, and just letting folks who specify
empty values deal with the fallout for programs that assume argv[0]
exists.